packages/realtime-js/src/Realtime.js

```diff
@@ -0,0 +1,70 @@
+import { Socket } from 'phoenix-channels'
+
+export default class Realtime {
+  constructor(apiSocket){
+    this.socket = new Socket(apiSocket)
+    this.channel = this.socket.channel('realtime')
+    this.listeners = []
+
+    this.socket.onOpen(() => {
+      console.log('Socket connected')
+    })
+    this.socket.onClose(() => {
+      console.log('Socket disconnected')
+    })
+  }
+
+  start(listeners) {
+    this.socket.connect()
+    if (this.channel.state !== 'joined'){
+      this.channel
+        .join()
+        .receive('ok', resp => console.log('Joined Realtime successfully', resp))
+        .receive('error', resp => console.log('Unable to join', resp))
+        .receive('timeout', () => console.log('Networking issue. Still waiting...'))
+
+      listeners.forEach( listener => {
+        this.addListener(listener)
+      })
+    }
+  }
+
+  stop(){
+    this.listeners.forEach(({ messageName, messagae }) => {
+      this.channel.off(messageName, message)
+    })
+    this.socket.disconnect()
+  }
+
+  /**
+   * takes in an object with keys
+   * - messageName
+   * - messgage 
+   * to:
+   * 1. add to this.listener
+   * 2. turn channel on
+   * @param {object} listener
```

Comment:
@dragarcia perhaps you can add 3 things to the JSdoc:

1. an` @example` 
2. change the @param to show the required items. i.e:
3. The description should be something like "Attaches a listener to the socket which gets fired on any change"

```
@param {object} listener Some description
@param {string} listener.channel Some description
@param {function} listener.callback A callback function to fire when the channel receives an change
```